### PR TITLE
Feat: new connect button + deprecate mobile pairing

### LINKED ISSUE #2540: Deprecate Mobile Pairing
Gradually deprecate the mobile pairing feature due to maintance costs:

* Remove the QR code from the wallet connection popup in the header
* Keep the mobile pairing option in the onboard modal, but show a 1-month notice about its deprecation

### PR DESCRIPTION
**Pull request by katspaugh**
## What it solves

Resolves #2540

## How this PR fixes it

Removes the mobile pairing from the Connect wallet popup, and the popup itself along with it. It will now show a Connect button directly in the header.
Screenshot 2023-09-25 at 17 14 16

This PR also adds a deprecation warning in the mobile pairing onboard modal. The wording and the date to be adjusted.

Screenshot 2023-09-25 at 17 13 51

### CODE REVIEW

File: src/components/common/PairingDetails/PairingDeprecationWarning.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Alert } from '@mui/material'
+
+const PairingDeprecationWarning = (): React.ReactElement => {
+  return (
+    <Alert severity="warning" sx={{ mb: 4 }}>
+      We&apos;re sorry to share that the Safe mobile pairing feature will be <b>discontinued</b> in November 2023.
```
Comment (iamacook) [inline src/components/common/PairingDetails/PairingDeprecationWarning.tsx:6]:
```suggestion
      We regret to announce that the Safe{Wallet} mobile pairing feature will be <b>discontinued</b> on 15th November 2023.
```

File: src/components/common/PairingDetails/PairingDeprecationWarning.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+      Please plan a migration to a different wallet as soon as possible.
```
Comment (usame-algan) [inline src/components/common/PairingDetails/PairingDeprecationWarning.tsx:7]:
```suggestion
      The Safe mobile pairing feature will be <b>discontinued</b> in November 2023.
      Please plan a migration to a different wallet as soon as possible.
```
I would suggest to keep it a bit shorter

Comment (iamacook) [inline src/components/common/PairingDetails/PairingDeprecationWarning.tsx:7]:
```suggestion
      Please migrate to a different wallet as soon as possible.
```

**Comment by TanyaEfremova** [comment]
@katspaugh Looks good to me!

**Review by iamacook — APPROVED** [review]

**Comment by katspaugh** [comment]
Since there's no logic changes in this PR, I'll merge and we'll test it on the RC branch.

**Pull request merged**